include/swift/SIL/SILBuilder.h

```diff
@@ -536,7 +536,8 @@ class SILBuilder {
 
   /// Create a `differentiable_function_extract` instruction with the given
   /// type. To be used only by LoweredByAddress.
-  DifferentiableFunctionExtractInst *createDifferentiableFunctionExtract(
+  DifferentiableFunctionExtractInst *
+  createDifferentiableFunctionExtractWithExplicitType(
```

Comment:
Why not just make the `SILType type` argument be optional?

Reply:
I created a distinct method to distinguish "creation with explicit type" because it is valid only in lowered SIL.

I don't feel strongly, so I'll change both the `SILBuilder` method and the `DifferentiableFunctionExtractInst` constructor to take an optional `SILType`.

Reply:
Done in https://github.com/apple/swift/pull/27923/commits/da3e68c5b72ec574683d9856c6561e861900c9b3.